src/item.cpp

```diff
@@ -6549,6 +6549,11 @@ bool item::can_contain( const item &it ) const
         // does the set of all sets contain itself?
         return false;
     }
+    // disallow putting portable holes into bags of holding
+    if( contents.bigger_on_the_inside( volume() ) &&
+        it.contents.bigger_on_the_inside( it.volume() ) ) {
+        return false;
+    }
```

Comment:
I feel like this ought to be implemented via a flag restriction on bags of holding to not contain `TARDIS` objects, rather than hardcoding it here (so that other mods *can* allow it if they really want to).  But that can be a later change, if at all.